File: DesignPatternDemo/Pattern/Response/ResponseMain.m
//
//  ResponseMain.m
//  DesignPatternDemo
//
//  Created by ruantong on 2019/5/11.
//  Copyright © 2019 wupeng. All rights reserved.
//

#import "ResponseMain.h"
#import "Women.h"
#import "Father.h"
#import "Husband.h"
#import "Son.h"

@implementation ResponseMain

- (void)main{
    NSMutableArray<id<IWomen>> * arrayList = [NSMutableArray new];
    for (int i = 0; i<5; i++) {
        [arrayList addObject:[[Women alloc] initWithType:arc4random()%3 request:@"我要出去逛街"]];
    }
    Father * father = [Father new];
    Husband * husband = [Husband new];
    Son * son = [Son new];
    [father setNext:husband];
    [husband setNext:son];
    
    for (id<IWomen> women in arrayList) {
        [father handleMessage:women];
    }
}

@end

File: DesignPatternDemo/Pattern/Response/ResponseMain.swift
//
//  ResponseMain.swift
//  DesignPatternDemo
//
//  Created by wupeng on 2020/3/5.
//  Copyright © 2020 wupeng. All rights reserved.
//

import Foundation

class ResponseMain: NSObject,IMain {
    func main() {
        var arrays:[IWomen] = [IWomen]()
        for _ in 1...5 {
            arrays.append(Women.init(type: RequestType(rawValue: NSInteger(arc4random()%3)) ?? RequestType.fatherType, request: "我要出去逛街"))
        }
        let father = Father()
        let husband = Husband()
        let son = Son()
        father.setNext(handler: husband)
        husband.setNext(handler: son)
        
        for women in arrays {
            father.HandleMessage(women: women)
        }
    }
}
